apps/somnia/src/commands/archive/weapon.ts

```diff
@@ -33,24 +34,43 @@ function getDropdown(id: string, refine: string) {
   return [
     new ActionRowBuilder().addComponents(
       new StringSelectMenuBuilder()
-        .setCustomId(`archive weapon ${id} ${refine}`)
+        .setCustomId(`archive weapon ${id} ${lang} ${refine}`)
         .setPlaceholder(`Refinement ${refine}`)
         .addOptions(options)
     ),
   ]
 }
 
-export function weaponArchive(
-  id: WeaponKey,
-  name: string,
-  data: any,
-  args: string
-) {
+export async function weaponArchive(id: WeaponKey, lang: string, args: string) {
+  const namespace = `weapon_${id}_gen`
+  await i18nInstance.loadNamespaces(namespace)
   const msg: any = {}
+  let name: string = translate(namespace, 'name', lang)
   let text = ''
-  //weapon rarity color
-  const rarity = getWeaponStat(id).rarity
+  const stat = getWeaponStat(id)
+  //mainstat
+  const ascension = stat.ascensionBonus[stat.mainStat.type] ?? [0]
+  const mainstat =
```

Comment:
Why not use Pando? Could be a future TODO as well i suppose, once we have more of the UI sheets and sheets done